### PR TITLE
Corrects Course Instructor visualizations

### LINKED ISSUE #2952: Instructional time (minutes) calculating incorrectly in visualizations
example:

go to https://ilios-frontend.netlify.app/courses/1936/sessions/64245.

The instructor "Lawrence Amanda Chapman" is assigned to two offerings, one with a duration of 30 minutes, the other with a duration of 2 hours (120 minutes).

If you navigate to the visualizations for the course, and open the instructor viz page https://ilios-frontend.netlify.app/data/courses/1936/instructors/13600, this instructor is listed for 120 minutes of total instruction time. And the session in question - the only one for this instructor in this course, btw ("Medication Safety, Reconciliation, and Improving Patient Adherence Workshop") displays a unit total of 4350 minutes of instructional time.

Since the session is small groups, with offerings across six time periods: 3 30-minute sessions, and 3 2-hour sessions, totalling 450 total minutes of instructional time -- 150 minutes for each of the 27 learner groups -- NONE of the numbers we display make any sense.

The total instructional time for a single instructor should be: the total minutes of each individual offering to which they are attached, added together.

The total instructional time for a given activity should be: the total minutes of each offering TIME BLOCK, for that session, added together. This is much like the "count as one" calculation we run for the CIR.

### PR DESCRIPTION
**Pull request by stopfstedt**
fixes #2952 

this _only_ fixes the instructor visualizations, other course visualizations will need to be reviewed and possible fixed as a follow-up.

### CODE REVIEW

**Comment by saschaben** [comment]
@stopfstedt the instructor time/percent donute seems on target, but the "terms" bar graphs appear to be counting the total course time associated with each, rather than the total instructor time associated with each. can you verify which it is? I *think* it should be total instructional time for each term for the selected instructor...

**Comment by stopfstedt** [comment]
> @stopfstedt the instructor time/percent donute seems on target, but the "terms" bar graphs appear to be counting the total course time associated with each, rather than the total instructor time associated with each. can you verify which it is? I _think_ it should be total instructional time for each term for the selected instructor...


I think I  can answer that only partially, let  me try:
- the _total instructional time_ the given instructor teaches (in that course) for the given topic is correct.
- i am less sure about the percentages, and therefore the length of the bars in the terms chart. 

image

if you add all the instructional time that each topic is taught by this instructor up, and then put this into relation to each individual term, then this is the correct output.

For example: Cultural Humility is taught for 255 mins. If you add up all the minutes for all terms, you get a sum total of 1110 mins.
255 is 23% of 1110, so that checks out. However, total instructional time is 405mins for this instructor. 
we arrive at the inflated total of 1110 mins, because sessions have been tagged with multiple terms, which is not taken into consideration.
So perhaps a different formula should be applied here? and/or a different chart could be used? I have no idea.

**Comment by saschaben** [comment]
I don't have an answer, but I think we can arrive at one. let's discuss.

**Comment by stopfstedt** [comment]
> I don't have an answer, but I think we can arrive at one. let's discuss.
agreed. here's a quick and dirty adjustment - taking the percentage against the total instructional time instead of the inflated aggregate. 

image

makes a lot more sense to me - 255 is 63% of 405.

**Comment by stopfstedt** [comment]
work in progress: 

playing around with the labeling of the terms bars

image


maybe this providing the context we need?

**Comment by stopfstedt** [comment]
@saschaben - i pushed the code change that calculates percentage against the total sum of instructional ("offerings") time for each term, rather than the total aggregate of instructional time of all terms. 
even without a labeling change, the charted data makes a lot more sense to me now. 
please have another look.

**Comment by stopfstedt** [comment]
@saschaben i changed the labeling to show minutes instead of percentages.

image

does this pass muster?

**Comment by saschaben** [comment]
@stopfstedt this DOES pass muster -- I really like the updates & the simplification of the labeling on the barchart. It makes a ton more intuitive sense now. Also the numbers are correct, which helps!

**Review by jrjohnson — CHANGES_REQUESTED** [review]

File: addon/components/course-visualize-instructor.js [hunk 1]
```diff
@@ -1,16 +1,53 @@
...
-  @restartableTask
-  *load() {
-    this.academicYearCrossesCalendarYearBoundaries = yield this.iliosConfig.itemFromConfig(
-      'academicYearCrossesCalendarYearBoundaries'
-    );
+  @use academicYearCrossesCalendarYearBoundaries = new ResolveAsyncValue(() => [
+    this.iliosConfig.itemFromConfig('academicYearCrossesCalendarYearBoundaries'),
+  ]);
+
+  @use _sessions = new ResolveAsyncValue(() => [this.args.course.sessions]);
```
[the remark this answers is not in the document]
Comment (stopfstedt) [reply addon/components/course-visualize-instructor.js:15]: :heavy_check_mark:

File: addon/components/visualizer-course-instructor-session-type.js [hunk 2]
```diff
@@ -15,33 +15,36 @@ export default class VisualizerCourseInstructorSessionType extends Component {
...
   @tracked tooltipTitle = null;
 
-  @use sessions = new ResolveAsyncValue(() => [this.args.course.sessions, []]);
-  @use sessionsWithSessionType = new AsyncProcess(() => [
-    this.getSessionsWithSessionsTypes.bind(this),
-    this.sessions,
-  ]);
-  async getSessionsWithSessionsTypes(sessions) {
-    return map(sessions.toArray(), async (session) => {
+  @use _sessions = new ResolveAsyncValue(() => [this.args.course.sessions]);
```
[the remark this answers is not in the document]
Comment (stopfstedt) [reply addon/components/visualizer-course-instructor-session-type.js:18]: :heavy_check_mark:

File: addon-test-support/ilios-common/page-objects/components/visualizer-course-instructor-term.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { collection, create, notHasClass } from 'ember-cli-page-object';
+
+const definition = {
+  scope: '[data-test-visualizer-course-instructor-term]',
+  isIcon: notHasClass('no-icon'),
```
[the remark this answers is not in the document]
Comment (stopfstedt) [reply addon-test-support/ilios-common/page-objects/components/visualizer-course-instructor-term.js:5]: agreed - it's awkward, but that's how it is currently. out of scope for this/should be improved as a follow-up item. 

https://github.com/ilios/common/pull/2955/files#diff-ba7bde41d71daf7d0b3489a0a0c85c689f245f1f9a5eff1821813943e8508520R2

**Review by jrjohnson — APPROVED** [review]
📈 📊 🎉

**Pull request merged**